frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -227,93 +451,25 @@ const Table: React.FC<TableProps> = ({
-          } ${
-            expandRow && expandedRows.includes(index)
-              ? 'has-child-expanded'
-              : ''
-          }`}
-          key={`index:${index}`}
-          style={rowStyles}
-          ref={index === preExpandRow ? expandRowRef : null}
-        >
-          <>
-            {expandRow &&
-            (item.isExpandable || item.isExpandable === undefined) ? (
-              <ExpandingCell
-                index={index}
-                expandedRows={expandedRows}
-                onExpand={onExpand}
-                onCollapse={onCollapse}
-                rowValues={item}
-                onClick={setExpandedRows}
-                isSelectedRow={currentSelectedKey === item.key}
-              />
-            ) : (
-              <td />
-            )}
-            {Object.entries(item)
-              .filter(([key]) => fields.includes(key))
-              .map(([key, value], rowIndex) => {
-                const columnInfo = columns.find(({ field }) => field === key);
-                const horAlign: TextAlignmentValues = columnInfo
-                  ? columnInfo.horAlign || DEFAULT_TEXT_ALIGNMENT
-                  : DEFAULT_TEXT_ALIGNMENT;
-                const width =
-                  columnInfo && columnInfo.width
-                    ? `${columnInfo.width}px`
-                    : DEFAULT_CELL_WIDTH;
-                const cellStyle = {
-                  width,
-                };
-                // TODO: Improve the typing of this
-                let cellContent: React.ReactNode | typeof value = value;
-                if (columnInfo && columnInfo.component) {
-                  cellContent = columnInfo.component(value, rowIndex, item);
-                }
-
-                return (
-                  <td
-                    className={`ams-table-cell ${getCellAlignmentClass(
-                      horAlign
-                    )}`}
-                    key={`index:${rowIndex}`}
-                    style={cellStyle}
-                  >
-                    {cellContent}
-                  </td>
-                );
-              })}
-          </>
-        </tr>
-        {expandRow ? (
-          <tr
-            className={`ams-table-expanded-row ${
-              expandedRows.includes(index) ? 'is-expanded' : ''
-            }`}
-            key={`expandedIndex:${index}`}
-          >
-            <td className="ams-table-cell">
-              {/* Placeholder for the collapse/expand cell */}
-            </td>
-            <td className="ams-table-cell" colSpan={fields.length + 1}>
-              {expandRow(item, index)}
-            </td>
-          </tr>
-        ) : null}
-      </React.Fragment>
-    ));
+    body = getTableRows(
+      data,
+      columns,
+      currentSelectedKey,
+      preExpandPanelKey,
+      rowStyles,
+      onExpand,
+      onCollapse,
+      expandRowRef,
+      expandedRows,
+      setExpandedRows,
+      formatChildrenData,
+      preExpandRightPanel,
+      0
```

Comment:
Another benefit of the object as an input is that this 0 would be more explicit about what it represents. It will be a better API